lib/bench.fz

```diff
@@ -0,0 +1,52 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature bench
+#
+#  Author: Michael Lill (michael.lill@tokiwa.software)
+#
+# -----------------------------------------------------------------------
+
+# benchmark f for nano_seconds
+# returns: iterations per second
+#
+bench<R>(f Function<R>, nano_seconds u64, warm_up_nano_seconds u64) f64 is
+  start := fuzion.std.nano_time
+
+  warm_up =>
+    do
+      f.call
+    until fuzion.std.nano_time - start > warm_up_nano_seconds
+
+  warm_up
+
+  iterations := for iter := u64.zero, iter + u64s.one
```

Comment:
why don't you just write
```
  iter := u64 0, iter + 1
```
The `zero`/`one` constants are basically for generic code in `integer` that should also work for user-defined numbers-like things like a matrix.